pkg/game/elements/placed_tile.go

```diff
@@ -39,51 +40,49 @@ func (pos *Position) UnmarshalText(text []byte) error {
 type MeepleType uint8
 
 const (
-	NormalMeeple MeepleType = iota
+	NoneMeeple MeepleType = iota
+	NormalMeeple
 
 	MeepleTypeCount int = iota
 )
 
-// represents a legal position (and rotation) of a tile on the board
-type TilePlacement struct {
-	tiles.Tile
-	Pos Position
+type TileWithMeeple struct {
+	Features  []PlacedFeature
+	HasShield bool
+	Building  building.Building
 }
 
-func (placement TilePlacement) Rotate(_ uint) TilePlacement {
-	panic("Rotate() not supported on TilePlacement")
+func (placement PlacedTile) Rotate(_ uint) PlacedTile {
+	panic("Rotate() not supported on PlacedTile")
 }
 
-// represents a legal position of a meeple on the tile
-type MeeplePlacement struct {
-	Side side.Side
-	Type MeepleType
+type PlacedFeature struct {
+	feature.Feature
+	MeepleType
+	playerID uint8
 }
 
-// represents a legal move (tile placement and meeple placement) on the board
-type LegalMove struct {
-	TilePlacement
-	// LegalMove always has a `Meeple`. Whether it is actually placed
-	// is determined by `MeeplePlacement.Side` which will be `None`, if it isn't.
-	Meeple MeeplePlacement
+func ToPlacedTile(tile tiles.Tile) PlacedTile {
+	features := []PlacedFeature{}
+	for _, n := range tile.Features {
+		features = append(features, PlacedFeature{n, NoneMeeple, 255})
```

Comment:
Probably out of scope for this PR, but I think we should change playerID type from uint8 to a dedicated type.
The 255 is kind of a magic number and something like playerID.NonePlayer would be more intuitive (as I assume that's the intent of 255 here)